crypto/fipsmodule/bn/internal.h

```diff
@@ -18,11 +18,15 @@
 
 #include <ring-core/base.h>
 
-#if defined(OPENSSL_X86_64) && defined(_MSC_VER) && !defined(__clang__)
+#if defined(_MSC_VER) && defined(OPENSSL_64_BIT) && !defined(__clang__)
 #pragma warning(push, 3)
 #include <intrin.h>
 #pragma warning(pop)
+#if defined(OPENSSL_X86_64)
 #pragma intrinsic(_umul128)
+#elif defined(OPENSSL_AARCH64)
+#pragma intrinsic(__umulh)
```

Comment:
I'm not sure why BoringSSL avoids doing this. My guess is they aren't testing AArch64 Windows builds using MSVC, but instead forcing the use of clang.